### PR TITLE
scene instance class

### PR DESCRIPTION
**Pull request by mariarzv**

### CODE REVIEW

**Comment by st235** [comment]
@mariarzv can you, please, add GeDi as `git submodule`? Then we will always a) have a `specific` commit of GeDi in the repo and b) have clean and transparent history.

File: src/accumulation/gedi_accumulator_strategy.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    """Provides a strategy that concatenates point clouds 'as is'.
+
+    The strategy assumes that the point clouds are already aligned
+    and no further transformation is required.
+    """
+
+    def on_merge(self,
+                 initial_point_cloud: np.ndarray[float],
+                 next_point_cloud: np.ndarray[float]) -> np.ndarray[float]:
+        return np.concatenate((initial_point_cloud, next_point_cloud), axis=1)
```
Comment (st235) [inline src/accumulation/gedi_accumulator_strategy.py]: we need to call `gedi` here.

File: README.md [hunk 2]
```diff
@@ -1 +1,21 @@
...
+1. Navigate to the root directory of the 'CourseWork3DDetection' repository. You can use the following command to change the directory:
+
+```bash
+%cd /content/CourseWork3DDetection
+```
+
+2. After completing these steps, the NuScenes Mini dataset should be available in the expected directory for use in your project:
+
+``` bash
+!mkdir -p ../data/sets/nuscenes
```
Comment (st235) [inline README.md:17]: I am using `temp` instead of `data` as `temp` is ignored by git.

File: src/utils/gedi_registration.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+import torch
+import numpy as np
+import open3d as o3d
+from gedi.gedi import GeDi
+
+def run_point_cloud_registration(movable_cloud: str,
```
Comment (st235) [inline src/utils/gedi_registration.py:6]: can we replace this method with the code below:

```python
def run_point_cloud_registration(movable_cloud: str,
                                 static_cloud: str,
                                 numpoints: int) -> o3d.geometry.PointCloud:
    # getting a pair of point clouds
    pcd_move = o3d.io.read_point_cloud(movable_cloud)
    pcd_stay = o3d.io.read_point_cloud(static_cloud)
    run_point_cloud_registration_o3d(pcd_move, pcd_stay, numpoints)
```

File: src/utils/conversion_helper.py
Comment (st235) [inline src/utils/conversion_helper.py:1]: what about calling the file `o3d_helpder`?

File: src/utils/conversion_helper.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import open3d as o3d
+import numpy as np
+
+def convert_to_o3d_pointcloud(points: np.ndarray):
```
Comment (st235) [inline src/utils/conversion_helper.py:4]: Let's add type to `np.array` and return type

```python
def convert_to_o3d_pointcloud(points: np.ndarray[float]) -> o3d.geometry.PointCloud
```

File: src/utils/gedi_registration.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    # applying estimated transformation
+    pcd_move.transform(est_result01.transformation)
+    o3d.visualization.draw_plotly([pcd_move, pcd_stay])
+
+    return pcd_move
+
+
+def run_point_cloud_registration_o3d(pcd_move: o3d.geometry.PointCloud,
+                                    pcd_stay: o3d.geometry.PointCloud,
+                                    numpoints: int):
```
Comment (st235) [inline src/utils/gedi_registration.py:90]: don't forget return type `->o3d.geometry.PointCloud`

**Review by st235 — COMMENTED** [review]
Can you, please, check the files in your CL and add a new line to the end of the file where the line is absent?

File: .gitmodules [hunk 6]
```diff
@@ -0,0 +1,3 @@
+[submodule "gedi"]
```
Comment (st235) [inline .gitmodules:1]: 👍

File: src/scene/scene_inst_class.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    def __str__(self):
+        return f"Instance ID: {self.instance_id}, Category: {self.category}, " \
+               f"Scene ID: {self.scene_id}, Frames: {self.frames}, " \
+               f"Transformations: {len(self.transformations)}"
+
+    def __repr__(self):
+        return self.__str__()
+
+
+class FrameDetection(SceneInstance):
```
Comment (st235) [inline src/scene/scene_inst_class.py:32]: what do you think about `FrameDescriptor`?
Let's separate these 2 classes and put them under separate files.

File: src/accumulation/gedi_accumulator_strategy.py (hunk 1 above)
Comment (mariarzv) [reply src/accumulation/gedi_accumulator_strategy.py]: all files will autopep in the next PR

File: src/utils/conversion_helper.py
Comment (mariarzv) [reply src/utils/conversion_helper.py:1]: done

File: src/scene/scene_inst_class.py (hunk 7 above)
Comment (mariarzv) [reply src/scene/scene_inst_class.py:32]: we agreed to not change the name. I will add a more detailed description in the next PR

File: README.md (hunk 2 above)
Comment (mariarzv) [reply README.md:17]: may change later

File: src/accumulation/gedi_accumulator_strategy.py (hunk 1 above)
Comment (st235) [reply src/accumulation/gedi_accumulator_strategy.py]: this is not about `autopep`. This comment is about necessity to call `GeDi` to concatenate 2 point clouds

File: src/utils/gedi_registration.py (hunk 3 above)
Comment (mariarzv) [reply src/utils/gedi_registration.py:6]: updated in the new PR

**Pull request merged**